code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1423,4 +1462,67 @@ var/list/robot_verbs_default = list(
 		if(2) //New Module
 			connected_ai << "<br><br><span class='notice'>NOTICE - Cyborg module change detected: [name] has loaded the [designation] module.</span><br>"
 		if(3) //New Name
-			connected_ai << "<br><br><span class='notice'>NOTICE - Cyborg reclassification detected: [oldname] is now designated as [newname].</span><br>"
\ No newline at end of file
+			connected_ai << "<br><br><span class='notice'>NOTICE - Cyborg reclassification detected: [oldname] is now designated as [newname].</span><br>"
+
+
+//**INTEGRATED LIGHTS STUFF: NEW TOGGLE_COMPONENT COMMAND, TOGGLE_LIGHTS
+/mob/living/silicon/robot/verb/toggle_component()
+	set category = "Robot Commands"
+	set name = "Toggle Component"
+	set desc = "Toggle a component, conserving power."
+
+	var/list/installed_components = list()
+	for(var/V in components)
+		if(V == "power cell") continue
+		var/datum/robot_component/C = components[V]
+		if(C.installed)
+			installed_components += V
+
+	var/toggle = input(src, "Which component do you want to toggle?", "Toggle Component") as null|anything in installed_components
+	if(!toggle)
+		return
+
+	var/datum/robot_component/C = components[toggle]
+	if(C.name == "integrated light")//integrated lights turn on when its component is turned on
+		toggle_lights()//call the toggle verb.
+	else if(C.toggled)
+		C.toggled = 0
+		src << "\red You disable your [C.name]."
+	else
+		C.toggled = 1
+		src << "\red You enable your [C.name]."
+
+// integrated lights
+/mob/living/silicon/robot/verb/toggle_lights()
+	set category = "Robot Commands"
+	set name = "Toggle Lights"
+	var/datum/robot_component/C = components["integrated light"]
+	if(!C.installed)
+		C.toggled = 0
+		set_light(0)
+		usr << "\blue You disable your integrated light."
+		src << "\red You do not have a functional integrated light."
+		return null
+	if(cell.charge<100 || !cell && !C.toggled)
+		src << "\red You do not have adequate power to utilize your integrated light."
+		return null
+	if(C.toggled)

```

Comment:
This is pointless, it should be one proc using the ? operator and C.toggled = !C.toggled